index.js

```diff
@@ -14,205 +14,221 @@ app.use(require("body-parser").json());
 app.use(express.static("public"));
 
 app.use((req, res, next) => {
-    const proto = req.headers["x-forwarded-proto"];
-    if (proto && proto !== "https") {
-        return res.redirect(302, `https://${req.hostname}${req.originalUrl}`);
-    }
-    
-    return next();
+  const proto = req.headers["x-forwarded-proto"];
+  if (proto && proto !== "https") {
+    return res.redirect(302, `https://${req.hostname}${req.originalUrl}`);
+  }
+
+  return next();
 });
 
 app.use((req, res, next) => {
-    res.setHeader("X-Frame-Options", "DENY");
-    next();
+  res.setHeader("X-Frame-Options", "DENY");
+  next();
 });
 
 const FileStore = require("session-file-store")(session);
 app.use(
-    session({
-        store: new FileStore(),
-        secret: process.env.JWT_SECRET,
-        resave: false,
-        saveUninitialized: false,
-        proxy: true,
-        cookie: { secure: process.env.NON_SECURE_SESSION !== "true" },
-    })
-    );
-    
+  session({
+    store: new FileStore(),
+    secret: process.env.JWT_SECRET,
+    resave: false,
+    saveUninitialized: false,
+    proxy: true,
+    cookie: { secure: process.env.NON_SECURE_SESSION !== "true" },
+  }),
+);
+
 app.set("view engine", "jade");
 
 const isJson = (str) => {
-    try {
-        JSON.parse(str);
-    } catch (e) {
-        return false;
-    }
-    return true;
-}
+  try {
+    JSON.parse(str);
+  } catch (e) {
+    return false;
+  }
+  return true;
+};
 const valid = new Validator();
 const discoverySchema = {
-    "type": "object",
-    "properties": {
-      "authorization_endpoint": {"type": "string"},
-      "token_endpoint": {"type": "string"},
-      "userinfo_endpoint": {"type": "string"},
-      "jwks_uri": {"type": "string"},
-    },
-    "required": ["authorization_endpoint","token_endpoint","userinfo_endpoint","jwks_uri"]
-  };
+  type: "object",
+  properties: {
+    authorization_endpoint: { type: "string", format: "uri" },
+    token_endpoint: { type: "string", format: "uri" },
+    userinfo_endpoint: { type: "string", format: "uri" },
+    jwks_uri: { type: "string", format: "uri" },
```

Comment:
@DanOnCall here we can specify the format of the string. If is not a valid URI, the validation will thrown an error 👍